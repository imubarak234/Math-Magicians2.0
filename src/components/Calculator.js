import React, { useState } from 'react';
import './calc.css';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [total, setTotal] = useState(null);
  const [next, setNext] = useState(null);
  const [operation, setOperation] = useState(null);
  const [displays, setDisplays] = useState('');

  const inputDisplay = (e) => {
    let ans = {};
    let ans2;
    const ans1 = {
      next,
      total,
      operation,
    };
    ans = calculate(ans1, e.target.getAttribute('name'));

    if (ans.next) ans2 = ans.next;
    else if (ans.total) ans2 = ans.total;
    else if (ans.operation) ans2 = ans.operation;
    else ans2 = 0;

    setTotal(ans.total);
    setNext(ans.next);
    setOperation(ans.operation);
    setDisplays(ans2);
  };

  return (
    <div className="tabs container d-flex">
      <div className="w-50 align-self-start">
        <h2 className="h1">Lets do Some Math!</h2>
      </div>
      <div className="w-50">
        <h2>Math Calculator</h2>
        <table>
          <colgroup>
            <col span="3" id="col-1" />
            <col span="1" id="col-2" />
          </colgroup>
          <tr>
            <th colSpan="4"><input type="text" placeholder="0" aria-label="o" value={displays} data-testid="input" /></th>
          </tr>
          <tr>
            <td><button type="button" name="AC" data-testid="AC" onClick={inputDisplay}>AC</button></td>
            <td><button type="button" name="+/-" data-testid="+/-" onClick={inputDisplay}>+/-</button></td>
            <td><button type="button" name="%" data-testid="%" onClick={inputDisplay}>%</button></td>
            <td><button type="button" name="รท" data-testid="รท" className="orange" onClick={inputDisplay}>รท</button></td>
          </tr>
          <tr>
            <td><button type="button" name="7" data-testid="7" onClick={inputDisplay}>7</button></td>
            <td><button type="button" name="8" data-testid="8" onClick={inputDisplay}>8</button></td>
            <td><button type="button" name="9" data-testid="9" onClick={inputDisplay}>9</button></td>
            <td><button type="button" name="x" data-testid="x" className="orange" onClick={inputDisplay}>x</button></td>
          </tr>
          <tr>
            <td><button type="button" name="4" data-testid="4" onClick={inputDisplay}>4</button></td>
            <td><button type="button" name="5" data-testid="5" onClick={inputDisplay}>5</button></td>
            <td><button type="button" name="6" data-testid="6" onClick={inputDisplay}>6</button></td>
            <td><button type="button" name="-" data-testid="-" className="orange" onClick={inputDisplay}>-</button></td>
          </tr>
          <tr>
            <td><button type="button" name="1" data-testid="1" onClick={inputDisplay}>1</button></td>
            <td><button type="button" name="2" data-testid="2" onClick={inputDisplay}>2</button></td>
            <td><button type="button" name="3" data-testid="3" onClick={inputDisplay}>3</button></td>
            <td><button type="button" name="+" data-testid="+" className="orange" onClick={inputDisplay}>+</button></td>
          </tr>
          <tr>
            <td colSpan="2"><button type="button" name="0" data-testid="0" onClick={inputDisplay}>0</button></td>
            <td><button type="button" name="." data-testid="." onClick={inputDisplay}>.</button></td>
            <td><button type="button" name="=" data-testid="=" className="orange" onClick={inputDisplay}>=</button></td>
          </tr>
        </table>
      </div>
    </div>
  );
};

export default Calculator;
