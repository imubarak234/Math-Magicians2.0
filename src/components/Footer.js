import React from "react";

const Footer = () => {
  return (
    <div className="footer">
      <footer className="fixed-bottom bg-dark py-5">
        <div className="container">
          <p className="text-white text-center">
          Copyright &copy; designed 2022
          </p>
        </div>
      </footer>
    </div>
  )
}

export default Footer;