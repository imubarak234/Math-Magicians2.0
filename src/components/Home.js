import React from 'react';

const Home = () => (
  <div className="container mb-5">
    <h2 className="h1 my-5">Welcome To Our Page!</h2>
    <p className="h5 my-5">
      Math Magicians is a single page application (SPA)
      built with React for people who love mathematics.
      It helps to make calculations straight forward and easy.
      Mathematics provides an effective way of building mental discipline and
      encourages logical reasoning and mental rigor.
      Autem alias laudantium natus totam! Adipisci, autem. Eius
      distinctio modi mollitia incidunt eum accusamus
      fugiat dolor eligendi illum, architecto autem assumenda, sequi
      similique esse quas molestiae deleniti veritatis earum
      maiores animi repellendus tempora consequatur suscipit dolore! Facilis!
    </p>

    <p className="h5 my">
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
      In vero dolores tenetur dignissimos? Laborum fuga sit
      maxime eum veritatis dolor ipsam dolorem dolores amet,
      officia distinctio minima consequuntur aliquam ratione,
      sapiente explicabo architecto.
      Autem alias laudantium natus totam! Adipisci, autem. Eius
      distinctio modi mollitia incidunt eum accusamus
      fugiat dolor eligendi illum, architecto autem assumenda, sequi
      similique esse quas molestiae deleniti veritatis earum
      maiores animi repellendus tempora consequatur suscipit dolore! Facilis!
    </p>
  </div>
);

export default Home;
