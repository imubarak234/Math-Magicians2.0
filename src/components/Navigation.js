import React from 'react';
import { NavLink } from 'react-router-dom';

const Navs = () => (
  <div className="navs">
    <nav className="navbar navbar-expand navbar-dark bg-dark">
      <div className="container">
        <NavLink className="navbar-brand" to="/">
          Math Magicians
        </NavLink>
        <div>
          <ul className="navbar-nav">
            <li className="nav-item">
              <NavLink className="nav-link" to="/">
                Home
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/calculator">
                Calculator
              </NavLink>
            </li>
            <li className="nav-item">
              <NavLink className="nav-link" to="/quotes">
                Quotes
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
);

export default Navs;
