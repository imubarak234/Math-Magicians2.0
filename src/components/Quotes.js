import React from "react";

const Quotes = () => {
  return (
    <div className="container">
      <p className="display-3 my-4"> Mathematics is not about numbers, equations, computattions, or algorithms: it 
      is about understanding. - William Paul Thurston</p>
      <p className="display-3 my-4"> Without mathematics, thereâ€™s nothing you can do. Everything around you is mathematics. Everything around you is numbers.
       - Shakuntala Devi, Indian writer and mental calculator</p>
      <p className="display-3 my-4"> Pure mathematics is, in its way, the poetry of logical ideas. - Albert Einstein, German theoretical physicist</p>
      <p className="display-3 my-4"> Mathematics knows no races or geographic boundaries; for mathematics, the cultural world is one country.
       - David Hilbert, German mathematician</p>
    </div>
  )
}

export default Quotes;