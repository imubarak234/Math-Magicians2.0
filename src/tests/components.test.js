import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Footer from '../components/Footer';
import Home from '../components/Home';
import Calculator from '../components/Calculator';
import Navigation from '../components/Navigation';
import Quotes from '../components/Quotes';

describe('Series of testing, rendering the various components', () => {
  it('renders Footer correctly', () => {
    const tree = renderer
      .create(<Footer />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Home correctly', () => {
    const tree = renderer
      .create(<Home />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Navigation correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter to="/" className="navbar-brand">
          <Navigation />
        </MemoryRouter>,

      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Calculator correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Quotes correctly', () => {
    const tree = renderer
      .create(<Quotes />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Series of tests for checking user inputs', () => {
  test('input display', () => {
    render(<Calculator />);

    expect(screen.getByTestId('input').value).toContain('');
  });

  test('Test button input 8', () => {
    render(<Calculator />);

    const kick = screen.getByTestId('8');
    fireEvent.click(kick);

    expect(screen.getByTestId('input').value).toContain('8');
  });

  test('Test button input +', () => {
    render(<Calculator />);

    const kick = screen.getByTestId('+');
    fireEvent.click(kick);

    expect(screen.getByTestId('input').value).toContain('+');
  });

  test('Test button input 8', () => {
    render(<Calculator />);

    const kick = screen.getByTestId('8');
    fireEvent.click(kick);
    fireEvent.click(kick);

    expect(screen.getByTestId('input').value).toContain('88');
  });

  test('Test button input -', () => {
    render(<Calculator />);

    const kick = screen.getByTestId('-');
    fireEvent.click(kick);

    expect(screen.getByTestId('input').value).toContain('-');
  });
});
