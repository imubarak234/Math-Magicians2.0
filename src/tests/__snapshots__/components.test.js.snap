// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Series of testing, rendering the various components renders Calculator correctly 1`] = `
<div
  className="tabs container d-flex"
>
  <div
    className="w-50 align-self-start"
  >
    <h2
      className="h1"
    >
      Lets do Some Math!
    </h2>
  </div>
  <div
    className="w-50"
  >
    <h2>
      Math Calculator
    </h2>
    <table>
      <colgroup>
        <col
          id="col-1"
          span="3"
        />
        <col
          id="col-2"
          span="1"
        />
      </colgroup>
      <tr>
        <th
          colSpan="4"
        >
          <input
            aria-label="o"
            data-testid="input"
            placeholder="0"
            type="text"
            value=""
          />
        </th>
      </tr>
      <tr>
        <td>
          <button
            data-testid="AC"
            name="AC"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
        </td>
        <td>
          <button
            data-testid="+/-"
            name="+/-"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
        </td>
        <td>
          <button
            data-testid="%"
            name="%"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </td>
        <td>
          <button
            className="orange"
            data-testid="÷"
            name="÷"
            onClick={[Function]}
            type="button"
          >
            ÷
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            data-testid="7"
            name="7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
        </td>
        <td>
          <button
            data-testid="8"
            name="8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
        </td>
        <td>
          <button
            data-testid="9"
            name="9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
        </td>
        <td>
          <button
            className="orange"
            data-testid="x"
            name="x"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            data-testid="4"
            name="4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
        </td>
        <td>
          <button
            data-testid="5"
            name="5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
        </td>
        <td>
          <button
            data-testid="6"
            name="6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
        </td>
        <td>
          <button
            className="orange"
            data-testid="-"
            name="-"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            data-testid="1"
            name="1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </td>
        <td>
          <button
            data-testid="2"
            name="2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
        </td>
        <td>
          <button
            data-testid="3"
            name="3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </td>
        <td>
          <button
            className="orange"
            data-testid="+"
            name="+"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          <button
            data-testid="0"
            name="0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
        </td>
        <td>
          <button
            data-testid="."
            name="."
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </td>
        <td>
          <button
            className="orange"
            data-testid="="
            name="="
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </td>
      </tr>
    </table>
  </div>
</div>
`;

exports[`Series of testing, rendering the various components renders Footer correctly 1`] = `
<div
  className="footer"
>
  <footer
    className="fixed-bottom bg-dark py-3"
  >
    <div
      className="container"
    >
      <p
        className="text-white text-center"
      >
        Copyright © designed 2022
      </p>
    </div>
  </footer>
</div>
`;

exports[`Series of testing, rendering the various components renders Home correctly 1`] = `
<div
  className="container mb-5"
>
  <h2
    className="h1 my-5"
  >
    Welcome TO our Page!
  </h2>
  <p
    className="h5 my-5"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. In vero dolores tenetur dignissimos? Laborum fuga sit maxime eum veritatis dolor ipsam dolorem dolores amet, officia distinctio minima consequuntur aliquam ratione, sapiente explicabo architecto. Autem alias laudantium natus totam! Adipisci, autem. Eius distinctio modi mollitia incidunt eum accusamus fugiat dolor eligendi illum, architecto autem assumenda, sequi similique esse quas molestiae deleniti veritatis earum maiores animi repellendus tempora consequatur suscipit dolore! Facilis!
  </p>
  <p
    className="h5 my"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit. In vero dolores tenetur dignissimos? Laborum fuga sit maxime eum veritatis dolor ipsam dolorem dolores amet, officia distinctio minima consequuntur aliquam ratione, sapiente explicabo architecto. Autem alias laudantium natus totam! Adipisci, autem. Eius distinctio modi mollitia incidunt eum accusamus fugiat dolor eligendi illum, architecto autem assumenda, sequi similique esse quas molestiae deleniti veritatis earum maiores animi repellendus tempora consequatur suscipit dolore! Facilis!
  </p>
</div>
`;

exports[`Series of testing, rendering the various components renders Navigation correctly 1`] = `
<div
  className="navs"
>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      className="container"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
      >
        Math Magicians
      </a>
      <div>
        <ul
          className="navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/calculator"
              onClick={[Function]}
            >
              Calculator
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="/quotes"
              onClick={[Function]}
            >
              Quotes
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;

exports[`Series of testing, rendering the various components renders Quotes correctly 1`] = `
<div
  className="container mb-5"
>
  <p
    className="h5 my-4"
  >
     
    Mathematics is not about numbers, equations, computattions, or algorithms: it is about understanding. - William Paul Thurston
  </p>
  <p
    className="h5 my-4"
  >
     
    Without mathematics, there’s nothing you can do. Everything around you is mathematics. Everything around you is numbers. - Shakuntala Devi, Indian writer and mental calculator
  </p>
  <p
    className="h5 my-4"
  >
     
    Pure mathematics is, in its way, the poetry of logical ideas. - Albert Einstein, German theoretical physicist
  </p>
  <p
    className="h5 my-4"
  >
     
    Mathematics knows no races or geographic boundaries; for mathematics, the cultural world is one country. - David Hilbert, German mathematician
  </p>
</div>
`;
